gh-99987: fcntl add the `F_KINFO` constant for FreeBSD 13.1 and onwards.

devnexen opened the pull request:
<!-- gh-issue-number: gh-99987 -->
* Issue: gh-99987
<!-- /gh-issue-number -->

devnexen committed:
fcntl add the `F_KINFO` constant for FreeBSD 13.1 and onwards.

blurb-it[bot] committed:
📜🤖 Added by blurb_it.

serhiy-storchaka reviewed (commented):
`fcntl()` does not contain code specific to other operations. The user is expected to provide a buffer object of the correct size and properly initialized.

devnexen committed:
changes from remarks.

- removing custom code out of `fcntl.fcntl`.
- F_GETPATH requires MAXPATHLEN or greater so we increase the buffer size
to be able to cater with F_KINFO greater size.
- `fcntl.kinfoalloc` preallocate a proper kinfo_file buffer.
- `fcntl.kinfodict` returns a handy representation of the data.

serhiy-storchaka commented on Doc/library/fcntl.rst:
```diff
@@ -57,6 +57,8 @@ descriptor.
    ``FICLONERANGE`` constants, which allow to share some data of one file with
    another file by reflinking on some filesystems (e.g., btrfs, OCFS2, and
    XFS). This behavior is commonly referred to as "copy-on-write".
+   On FreeBSD >= 13.1, the :mod:`fcntl` module exposes the ``F_KINFO``
```
It should be `versionchanged:: 3.13` now.

serhiy-storchaka reviewed (commented):
I understand why do you want to add `kinfoalloc()` and `kinfodict()`, but for now there are no precedents for other operations. See for example the use of `F_SETLK` in tests and in third-party examples. They use `struct.pack()` and `struct.unpack()`. The format string, unfortunately, is platform depending.

If we came with better solution, it should be universal, and cover all other `fcntl()` and `ioctl()` operations. It is a different and much larger issue. Let limit this issue for adding `F_KINFO` (and maybe other constants?).
